### DISCUSSION TITLE
making-authenticated-cross-origin-requests-with-aspnetcore-identity/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#378] [Announcements]
# making-authenticated-cross-origin-requests-with-aspnetcore-identity/

In this post I describe how to make authenticated cross-origin requests from the browser to an app using ASP.NET Core Identity.

https://andrewlock.net/making-authenticated-cross-origin-requests-with-aspnetcore-identity/

**Comment by sherif-botros** [comment]
Thanks for this 🙌 Great post, as usual.
Question, above when you say _"... browsers make a request as normal, but before passing the response to the JavaScript code that initiated the request, it checks ..."_ does this mean that the response, in full is received by the browser, and can be sniffed or gotten-to if the internal of the browser execution is available?

**Reply by andrewlock** [reply]
Yes, this is a _browser_ control - you can call the API outside the context of a browser and CORS doesn't apply at all. The request is sent across the wire, but the browser doesn't expose it.

In reality, that's not a big concern these days. If an attacker already has compromised the browser then all bets are off anyway!

**Comment by Ninjanaut** [comment]
Hello Andrew, great article! I would like to ask, it's possible (for development) to use IIS Express with following settings
```json
"windowsAuthentication": true,
"anonymousAuthentication": false,
 ```
in  `launchSettings.json` and to allow preflight at the same time? I have issue that POST request fails because preflight request is blocked by the browser. When I set `"anonymousAuthentication": true` the preflight is not blocked but then the user is not authenticated at all (missing HttpContext.User.Identity values).

**Reply by Ninjanaut** [reply]
The error is following `Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

**Comment by maojsm** [comment]
Muito obrigado, salvou meu dia.
